msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -201,6 +201,10 @@ public static DeviceCodeFlowCommandParameters createDeviceCodeFlowCommandParamet
             @NonNull String[] scopes){
 
         // TODO: Consider implementing support for PoP
+
+        final String uri = ((AzureActiveDirectoryAuthority) configuration.getAuthorities().get(0)).getAudience().getCloudUrl();
+        final Authority authority = Authority.getAuthorityFromAuthorityUrl(uri);
+
```

Comment:
Why can't we just use default authority?

Reply:
Default authority has "common" as tenantID

Reply:
Default authority would be whatever was declared default in the config....or if there is only one declared then it would be that one authority. It would have `common` if the dev put common in the config, otherwise it won't. If it does have common, why is that a problem?

Reply:
This is my config:
![image](https://user-images.githubusercontent.com/65684798/89474486-a3899800-d73a-11ea-9836-1fcbc34aa26a.png)

Common as the tenantID is what's causing /common to be added to the end of the authority url later down the line. This issue doesn't occur in acquireToken() because the command parameters for that also use a custom authority.

Reply:
refer to `getAuthorityUri()` in `AzureActiveDirectoryAuthroity`, this is where the tenant is appended to the end of the authority.

Reply:
Okay, I'm not completely clear on this. Let's sync up offline.

Reply:
Synced offline with @t-fadura. There's a bug in MSAL and has been for quite some time and that's what's causing this issue. Created GitHub issue here: https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/1121